ascent/examples/ascent_if_let_clause.rs

```diff
@@ -0,0 +1,42 @@
+//! Conditional `if let` clause
+
+use ascent::ascent;
+
+ascent! {
+    // Facts:
+
+    relation option(Option<isize>);
+    
+    // Rules:
+    
+    relation some(isize);
+
+    some(y) <-- option(x), if let Some(y) = x;
+
+    // The rule above could alternatively also be written
+    // using the following short-hand syntax:
+    // some(y) <-- option(?Some(y));
```

Comment:
I think it'd be better to uncomment this. This would be a redundant rule, but I think it's fine for an example showing Ascent features.

Reply:
Done.